annotto-front/src/modules/project/components/common/image/ZoomMark.js

```diff
@@ -0,0 +1,210 @@
+import { flatten, isNumber, min, max } from 'lodash'
+import PropTypes from 'prop-types'
+import { useMemo, useRef, useState, useEffect } from 'react'
+import { Group, Image, Line, Text, Transformer } from 'react-konva'
+import useImage from 'use-image'
+
+import theme from '__theme__'
+
+const ZoomMark = ({
+  index,
+  tasks,
+  annotationIndex,
+  predictionIndex,
+  zone,
+  value,
+  imageWidth,
+  imageHeight,
+  isSelected,
+  onValidateClick,
+  onDeleteClick,
+  onSelectClick,
+  onTransformEnd,
+  ...props
+}) => {
+  const lineRef = useRef()
+  const trRef = useRef()
+
+  const [isHovered, setIsHovered] = useState(false)
+
+  const [tagIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/tag.svg`)
+  const [patternIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/pattern.svg`)
+  const [closeIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/close.svg`)
+  const [validateIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/validate.svg`)
+
+  const isPrefill = isNumber(annotationIndex) && isNumber(predictionIndex)
+  const isPrediction = isNumber(predictionIndex) && !isNumber(annotationIndex)
+  const isEditable = !isPrediction && !isPrefill
+
+  const points = useMemo(
+    () => flatten(zone.map(({ x, y }) => [x * imageWidth, y * imageHeight])),
+    [zone, imageHeight, imageWidth]
+  )
+
+  const task = useMemo(() => {
```

Comment:
maybe `Task` could be passed as a props here to avoid passing down the whole `tasks` object? This could also avoid caching this variable?